SwiftCompilerSources/Sources/Optimizer/FunctionPasses/DeadStoreElimination.swift

```diff
@@ -75,7 +75,7 @@ let deadStoreElimination = FunctionPass(name: "dead-store-elimination") {
 }
 
 private func tryEliminate(store: StoreInst, complexityBudget: inout Int, _ context: FunctionPassContext) {
-  if !store.hasValidOwnershipForDeadStoreElimination {
+  if !store.hasValidOwnershipForDeadStoreElimination || !store.source.type.shouldExpand(context) {
```

Comment:
Can you add a short comment why preventing dead-store elimination of not-shouldExpand-types is beneficial?